skimage/feature/tests/test_sift.py

```diff
@@ -129,3 +129,18 @@ def test_no_descriptors_extracted_sift():
     detector_extractor = SIFT()
     with pytest.raises(RuntimeError):
         detector_extractor.detect_and_extract(img)
+
+def test_features_dtype():
+    detector_extractor = SIFT()
+    positions_oct = np.array([[1.0, 1.0]])
+    sigmas_oct = np.array([0.8])
+    octaves = np.array([0])
+    scales_oct = np.array([1])
+    gaussian_scalespace = detector_extractor._create_scalespace(img)
+    detector_extractor._compute_orientation(positions_oct, scales_oct, sigmas_oct,
```

Comment:
This test uses internals that aren't part of the public API. Can the behavior you are addressing in this PR be reproduced by only using the public API of SIFT?

Reply:
I actually tried it through the public API, but it's just impossible to make a test image to reproduce this type instability issue. That's why I used the non-public API.

As one of the users, I personally wanted all of these non-public APIs to be the public APIs, though. If they were exposed, it would have been easier for me to do a bit non-standard thing. Could they be actually exposed?